### PR TITLE
♻️ Refactor/use grid

### LINKED ISSUE #165: Replace Grid component with new UI Kit grid system
With the introduction of the grid in the UI Kit, we should refactor anything that uses the Grid component and remove it completely.

### PR DESCRIPTION
**Pull request by XuTheBunny**
1. Study list: responsive layout (4, 3, 2, 1 column of cards according to screen size)
2. File annotation: responsive layout (stack "File description" and "Select a file type" on the small screen)
3. Study header: "study name" - cell-10, "contacts" - cell-2
4. File list: responsive layout (stack badges to the bottom if not on big screen)
5. Upload box: passing `className` as prop to `FileUploadTarget` component, move styles to css file, taking 8 cells and start from cell 3 (in File list view)
6. File list view and File annotation view: align components with study header text
7.  Study header loading: make loading stage the same div height, so that the page won't jump after populated.
8. App header: use grid system sizing so that it aligns with other components

Closes #165

### CODE REVIEW

**Comment by bdolly** [comment]
@XuTheBunny can you rename this branch to `refactor/use-grid` to help keep the branches tidy

File: src/components/FileAnnotation/FileEditor.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, {useState} from 'react';
 import PropTypes from 'prop-types';
 import {withRouter} from 'react-router-dom';
-import {Button, Icon} from 'kf-uikit';
+import {Button, Icon, GridContainer} from 'kf-uikit';
```
Comment (bdolly) [inline src/components/FileAnnotation/FileEditor.js]: 😃 seeing this makes me happy

File: src/views/AnnotationView.js [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
   const fileId = match.params.fileId;
   return (
-    <div className="sm:px-20 p-2 BodyContent">
-      <h3 className="text-blue font-normal my-16">Tell us about your files</h3>
-      <p className="pb-20">
+    <GridContainer collapsed="cells" className="my-20 px-12">
+      <h3 className="text-blue font-normal m-0 cell-12 row-1">
+        Tell us about your files
+      </h3>
+      <p className="m-0 cell-8 row-2">
```
Comment (bdolly) [inline src/views/AnnotationView.js]: Loving the file editor view on mobile, just need to make sure the help text goes full width on small screens 

--- 
`320px wide`
Screen Shot 2019-04-30 at 1 05 56 PM

File: src/views/NavBarView.js [hunk 3]
```diff
@@ -1,19 +1,17 @@
 import React from 'react';
 import {Query} from 'react-apollo';
 import {GET_STUDY_BY_ID} from '../state/queries';
-import {LoadingPlaceholder} from '../components/Loading';
```
Comment (bdolly) [inline src/views/NavBarView.js]: Did the study header loading state get added in this PR? If so can we move those changes to another PR

File: src/components/StudyHeader/StudyHeader.jsx [hunk 4]
```diff
@@ -3,54 +3,60 @@ import propTypes from 'prop-types';
...
+            last updated: {modifiedAt && <TimeAgo date={modifiedAt} />}
+          </small>
+        </p>
+        {loading ? (
+          <p className="titleText text-grey">Loading Study ...</p>
+        ) : (
+          <p className="titleText text-blue">{shortName || studyName}</p>
+        )}
+        {!loading && (
+          <div className="row-2 cell-2">
```
[the remark this answers is not in the document]
Comment (XuTheBunny) [reply src/components/StudyHeader/StudyHeader.jsx]: image

File: src/components/StudyHeader/StudyHeader.jsx [hunk 5]
```diff
@@ -3,54 +3,60 @@ import propTypes from 'prop-types';
...
+        )}
+      </GridContainer>
     </div>
   );
 };
 
 StudyHeader.propTypes = {
   /** the kf_id (SD_XXXXXXX) for the study  */
-  kfId: propTypes.string.isRequired,
+  kfId: propTypes.string,
```
[the remark this answers is not in the document]
Comment (XuTheBunny) [reply src/components/StudyHeader/StudyHeader.jsx]: This is for handling the "loading" status. When loading, the study header would show another look for loading, and once data comes in, it would rerender with the data.

File: src/components/StudyHeader/StudyHeader.jsx [hunk 6]
```diff
@@ -11,40 +12,35 @@ const StudyHeader = ({
...
-            <span className="study-modified-date pl-20 font-light">
-              <small>
-                last updated: <TimeAgo date={modifiedAt} />
-              </small>
-            </span>
-          </h3>
-          <h2 className="mt-4 font-light text-blue text-4xl">
-            {shortName || studyName}
-          </h2>
+    <div className={studyHeaderClass}>
```
Comment (dankolbman) [inline src/components/StudyHeader/StudyHeader.jsx]: Is it possible to remove this div and put the class name on the GridContainer to avoid another <div>?

File: src/components/StudyHeader/StudyHeader.jsx [hunk 7]
```diff
@@ -11,40 +12,35 @@ const StudyHeader = ({
...
   );
 };
 
 StudyHeader.propTypes = {
   /** the kf_id (SD_XXXXXXX) for the study  */
-  kfId: propTypes.string.isRequired,
+  kfId: propTypes.string,
   /** datetime of when the study was last updated */
-  modifiedAt: propTypes.string.isRequired,
+  modifiedAt: propTypes.string,
```
Comment (dankolbman) [inline src/components/StudyHeader/StudyHeader.jsx]: Why the sudden change in requirements?

File: src/components/FileList/FileElement.js [hunk 8]
```diff
@@ -87,56 +88,60 @@ const FileElement = ({
...
-            title={`Study ID: ${match.params.kfId}`}
-          >
-            {fileKfID} <CopyButton text={fileKfID} className={buttonClass} />
-          </span>
-          <span className={fileStatusClass}>
-            <Link
-              to={`/study/${match.params.kfId}/files/${fileKfID}`}
-              className={buttonClass}
+      <GridContainer collapsed="rows">
+        <div className="lg:cell-2 lg:row-1 row-2 cell-12 FileList--ElementBadge">
```
Comment (dankolbman) [inline src/components/FileList/FileElement.js]: Can all those classnames be moved into the `FileList--ElementBadge` definition?

File: src/containers/UploadContainer.js [hunk 9]
```diff
@@ -77,6 +77,7 @@ const UploadContainer = props => {
     >
       {(createFile, {data, error}) => (
         <FileUploadTarget
+          className={props.className}
```
Comment (dankolbman) [inline src/containers/UploadContainer.js]: I don't think that the `UploadContainer` has a `className` prop. (We should specify `prop-types` though)

File: src/components/StudyList/StudyList.jsx [hunk 10]
```diff
@@ -3,34 +3,32 @@ import PropTypes from 'prop-types';
...
-      </h1>
-      <ul className={studyListClass}>
+    <div className="bg-lightGrey min-h-screen py-32">
+      <GridContainer className={studyListClass}>
+        <h1 className="text-blue font-title row-1 cell-12">{title}</h1>
         {loading
           ? loadingCount.map(n => (
-              <li
-                className="p-8 cursor-not-allowed w-full sm:w-1/2 md:w-1/3 lg:w-1/4"
+              <div
```
[the remark this answers is not in the document]
Comment (XuTheBunny) [reply src/components/StudyList/StudyList.jsx]: For study list I changed from `<ul>..<li>` structure to `<GridContainer>..<div>`
